tsdb/head_append.go

```diff
@@ -358,6 +383,38 @@ func (a *headAppender) AppendExemplar(ref storage.SeriesRef, lset labels.Labels,
 	return storage.SeriesRef(s.ref), nil
 }
 
+func (a *headAppender) AppendMetadata(ref storage.SeriesRef, lset labels.Labels, meta metadata.Metadata) (storage.SeriesRef, error) {
```

Comment:
we should include a comment here like the one for `AppendExemplar` stating that for the head appender we don't create the series if it doesn't exist